### PR TITLE
feature: add asset on TicketRecurrent to add them on the ticket created  or create 1 ticket per asset

### PR DESCRIPTION
**Pull request by delcroip**
new version #9567 that is using GLPI 10 file names and structures

Task to be done:
- migrate methods from ticket to commonItilObject : done
- updating the Item_TicketRecurrent/ TicketRecurrent to add devices support:  Done
- adding the strategy setting (1 ticket for all or 1 ticket per device) in TicketRecurent: done
- adding the device in the created ticket according to strategy: Done
- manual testing strategy 1 item per ticket: passed
- manual testing all item on 1 ticket:  passed
- adding tests : TBD

NB: ChangeRecurrent won't be done in that PR

### CODE REVIEW

File: src/Item_Ticket.php [hunk 1]
```diff
@@ -368,894 +154,20 @@ public static function showItemToAdd($tickets_id, $itemtype, $items_id, $options
      *
      * @return void
      **/
-    public static function showForTicket(Ticket $ticket)
+    public static function showForObject($ticket, $options = [])
```
Comment (cedric-anne) [inline src/Item_Ticket.php:157]: Proposed PR will probably be integrated in GLPI 10.1, which is an intermediate version (it is too late for GLPI 10.0).
In intermediate version, we try to not introduce BC breaks., so, if you want to rename a method, you have to handle a deprecation state.

For instance:
```suggestion
    public static function showForTicket(Ticket $ticket)
    {
        Toolbox::deprecated();
        static::showForObject($ticket);
    }

    public static function showForObject($ticket, $options = [])
```

**Comment by delcroip** [comment]
Side remark, I have issue to generate automatic action/recurrent, it eventually create the ticket for the recurrent ticket but not every hours as I configured the recurrent ticket, could be linked to my test bed (docker) but still odd (next creation date is ok)

**Comment by cedric-anne** [comment]
> Side remark, I have issue to generate automatic action/recurrent, it eventually create the ticket for the recurrent ticket but not every hours as I configured the recurrent ticket, could be linked to my test bed (docker) but still odd (next creation date is ok)

You have to ensure that you PHP default timezone is same as your MySQL default timezone. There is a known issue that is not yet fixed.

**Comment by delcroip** [comment]
a question: should ajax/object_item.php be rename ajax/commonItilObject_item.php ?

**Comment by delcroip** [comment]
I have no Idea why the CI fails

**Comment by cedric-anne** [comment]
> I have no Idea why the CI fails

This was due to lack of strictness in changes you made in install/update scripts. I fixed it.

**Comment by delcroip** [comment]
thanks, when it will work I will rebase / apply it as patch to avoid the tons of commits (because of the merge I cannot squatch them)

**Comment by delcroip** [comment]
repalced by #11304

**Comment by cedric-anne** [comment]
When you want to synchronize with main branch, you should use `git pull --rebase`. With this, you will ensure that no merge commit is created, so squashing commit would be easier.

**Pull request closed without merging**